src/utils.rs

```diff
@@ -0,0 +1,114 @@
+use blake2::digest::{Update, VariableOutput};
+use blake2::Blake2bVar;
+use std::cmp::min;
+
+// Oracles
+
+/// Takes as input a hash and inverse probability $np$ and returns a
```

Comment:
inverse probability?

Reply:
Let's rename the function appropriately and fix the comment appropriately. Also, I'd briefly explain how the hash is used.

Reply:
The "inverse proba" was because of an error of mine when I used this oracle for H2 with parameter 1/q. Updated.

Reply:
What is wrong with the function name?